### PR TITLE
Ordering with Agora outside the US

### PR DESCRIPTION
**Pull request by FredericEspiau**
### Applicable Issues
MANAGER-1979

### CODE REVIEW

File: client/app/dedicatedCloud/datacenter/backup/dedicatedCloud-datacenter-backup.controller.js [hunk 1]
```diff
@@ -1,37 +1,52 @@
...
-  };
-
-  $scope.loadVeeam();
-});
+angular
+  .module('App')
+  .controller(
+    'DedicatedCloudSubDatacenterVeeamCtrl',
+    class {
+      constructor(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/datacenter/backup/dedicatedCloud-datacenter-backup.controller.js]:
```suggestion
      /* @ngInject */
      constructor(
```

File: client/app/dedicatedCloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js [hunk 2]
```diff
@@ -1,47 +1,113 @@
...
-            _.set(dc, 'consumption', null);
-          }
-          return dc;
-        });
+angular
+  .module('App')
+  .controller(
+    'DedicatedCloudSubDatacentersDatastoreCtrl',
+    class {
+      constructor(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js]:
```suggestion
      /* @ngInject */
      constructor(
```

File: client/app/dedicatedCloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js [hunk 3]
```diff
@@ -1,47 +1,113 @@
...
+              .all(datastores.map((dc) => {
+                if (dc.billing === 'HOURLY') {
+                  return this.DedicatedCloud
+                    .getDatastoreHourlyConsumption(
+                      this.$stateParams.productId,
+                      this.$stateParams.datacenterId,
+                      dc.id,
+                    )
+                    .then((response) => {
+                      _.set(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js]: IMO we are losing in readability.

File: client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js [hunk 4]
```diff
@@ -1,50 +1,92 @@
...
-          .getHostHourlyConsumption(
-            $stateParams.productId,
-            $stateParams.datacenterId,
-            host.hostId,
+angular
+  .module('App')
+  .controller(
+    'DedicatedCloudSubDatacentersHostCtrl',
+    class {
+      constructor(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/datacenter/host/dedicatedCloud-datacenter-host.controller.js]:
```suggestion
      /* @ngInject */
      constructor(
```

File: client/app/dedicatedCloud/datacenters/dedicatedCloud-datacenters.controller.js [hunk 5]
```diff
@@ -1,61 +1,104 @@
...
-          this.$scope.resetAction();
-          this.$scope.setMessage(this.$translate.instant('dedicatedCloud_datacenters_loading_error'), {
-            message: err.message,
-            type: 'ERROR',
+angular
+  .module('App')
+  .controller(
+    'DedicatedCloudDatacentersCtrl',
+    class {
+      constructor(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/datacenters/dedicatedCloud-datacenters.controller.js]:
```suggestion
      /* @ngInject */
      constructor(
```

File: client/app/dedicatedCloud/dedicatedCloud.controller.js [hunk 6]
```diff
@@ -1,273 +1,321 @@
...
-            };
-          },
-        },
-      });
+angular
+  .module('App')
+  .controller(
+    'DedicatedCloudCtrl',
+    class {
+      constructor(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/dedicatedCloud.controller.js]:
```suggestion
      /* @ngInject */
      constructor(
```

File: client/app/dedicatedCloud/user/delete/dedicatedCloud-user-delete.controller.js [hunk 7]
```diff
@@ -1,17 +1,55 @@
...
-      },
-    );
-  };
-});
+angular
+  .module('App')
+  .controller(
+    'DedicatedCloudUserDeleteCtrl',
+    class {
+      constructor(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/user/delete/dedicatedCloud-user-delete.controller.js]:
```suggestion
      /* @ngInject */
      constructor(
```

File: client/app/dedicatedCloud/license/dedicatedCloud-license.controller.js [hunk 8]
```diff
@@ -1,45 +1,85 @@
...
-  };
-
-  $scope.loadLicences();
-});
+angular
+  .module('App')
+  .controller(
+    'DedicatedCloudLicencesCtrl',
+    class {
+      constructor(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/license/dedicatedCloud-license.controller.js]:
```suggestion
      /* @ngInject */
      constructor(
```

File: client/app/dedicatedCloud/dedicatedCloud.service.js [hunk 9]
```diff
@@ -1,1313 +1,1929 @@
...
+        return this.OvhHttp.post(
+          '/dedicatedCloud/{serviceName}/upgradeHypervisor',
+          {
+            rootPath: 'apiv6',
+            urlParams: { serviceName },
+          },
+        );
+      }
+
+      isOptionT
```
Comment (antleblanc) [inline client/app/dedicatedCloud/dedicatedCloud.service.js]: Could be turned into a template string.

Comment (antleblanc) [inline client/app/dedicatedCloud/dedicatedCloud.service.js]: Could be turned into a template string.

File: client/app/dedicatedCloud/dedicatedCloud.service.js [hunk 10]
```diff
@@ -1,1313 +1,1929 @@
...
-        rootPath: 'apiv6',
-        urlParams: {
-          serviceName,
-        },
-      });
-    };
+  .service(
+    'DedicatedCloud',
+    class {
+      constructor(
```
Comment (antleblanc) [inline client/app/dedicatedCloud/dedicatedCloud.service.js]:
```suggestion
      /* @ngInject */
      constructor(
```

File: client/app/dedicatedCloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js (hunk 3 above)
Comment (FredericEspiau) [reply client/app/dedicatedCloud/datacenter/datastore/dedicatedCloud-datacenter-datastore.controller.js]: Indeed

**Comment by FredericEspiau** [comment]
@JeremyDec all done

**Comment by jleveugle** [comment]
Hello @FredericEspiau,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**